enum_methods.rb

```diff
@@ -0,0 +1,146 @@
+module Enumerable
+  def my_each
+    if block_given?
+      for i in self
+        yield(i)
+      end
+    else
+      puts 'No Block Given'
+    end
+  end
+
+  def my_each_with_index
+    count = 0
+    if block_given?
+      for i in self
+        yield(i, count)
+        count += 1
+      end
+    else
+      puts 'No Block Given'
+    end
+  end
+
+  def my_select
+    if instance_of?(Hash)
+      result = {}
+      my_each { |key, value| result[key] = value if yield(key, value) == true }
+    else
+      result = []
+      my_each { |value| result.push(value) if yield(value) == true }
+    end
+    result
+  end
+
+  def my_all?
+    count = 0
+    my_each { |value| count += 1 if yield(value) == true }
+    count == length
+  end
+
+  def my_any?
+    count = 0
+    my_each { |value| break count += 1 if yield(value) == true }
+    count.positive?
+  end
```

Comment:
- [ ] my_any? when no block or argument is given returns true if at least one of the collection is not false or nil

- [ ] my_any? when no block or argument is given returns false if at least one of the collection is not true

- [ ] my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class::Numeric

- [ ] my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class::Integer

- [ ] my_any? when a Regex is passed as an argument returns true if any of the collection matches the Regex

- [ ] my_any? when a Regex is passed as an argument returns false if none of the collection matches the Regex

- [ ] my_any? when a pattern other than Regex or a Class is given returns true if any of the collection matches the pattern